Signal/src/UIView+OWS.m

```diff
@@ -0,0 +1,170 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+#import "PureLayout.h"
+#import "UIView+OWS.h"
+
+// TODO: We'll eventually want to promote these into an OWSMath.h header.
+static inline CGFloat Clamp(CGFloat value, CGFloat minValue, CGFloat maxValue)
+{
+    return MAX(minValue, MIN(maxValue, value));
+}
+
+static inline CGFloat Clamp01(CGFloat value)
+{
+    return Clamp(value, 0.f, 1.f);
+}
+
+static inline CGFloat CGFloatLerp(CGFloat left, CGFloat right, CGFloat alpha)
+{
+    alpha = Clamp01(alpha);
+
+    return (left * (1.f - alpha)) + (right * alpha);
+}
+
+static inline CGFloat CGFloatInverseLerp(CGFloat value, CGFloat minValue, CGFloat maxValue)
+{
+    return (value - minValue) / (maxValue - minValue);
+}
+
+static inline CGFloat ScreenShortDimension()
+{
+    return MIN([UIScreen mainScreen].bounds.size.width, [UIScreen mainScreen].bounds.size.height);
+}
+
+static const CGFloat kIPhone5ScreenWidth = 320.f;
+static const CGFloat kIPhone7PlusScreenWidth = 414.f;
+
+CGFloat ScaleFromIPhone5To7Plus(CGFloat iPhone5Value, CGFloat iPhone7PlusValue)
+{
+    CGFloat screenShortDimension = ScreenShortDimension();
+    return round(CGFloatLerp(iPhone5Value,
+        iPhone7PlusValue,
+        CGFloatInverseLerp(screenShortDimension, kIPhone5ScreenWidth, kIPhone7PlusScreenWidth)));
+}
+
+CGFloat ScaleFromIPhone5(CGFloat iPhone5Value)
+{
+    CGFloat screenShortDimension = ScreenShortDimension();
+    return round(iPhone5Value * screenShortDimension / kIPhone5ScreenWidth);
+}
+
+#pragma mark -
+
+@implementation UIView (OWS)
```

Comment:
typically with categories I add our prefix to the methods, as in `ows_autoPinWidthToSuperview`, but these methods will be used so often, and are unlikely to clobber anything, so it seems worth breaking that convention.